libs/sdk-core/src/persist/swap.rs

```diff
@@ -94,16 +95,17 @@ impl SqliteStorage {
         &self,
         bitcoin_address: String,
         paid_msat: u64,
-    ) -> PersistResult<()> {
+        status: SwapStatus,
+    ) -> PersistResult<SwapInfo> {
         self.get_connection()?.execute(
-            "UPDATE swaps_info SET paid_msat=:paid_msat where bitcoin_address=:bitcoin_address",
+            "UPDATE swaps_info SET paid_msat=:paid_msat, status=:status where bitcoin_address=:bitcoin_address",
             named_params! {
              ":paid_msat": paid_msat,
              ":bitcoin_address": bitcoin_address,
+             ":status": status as u32,
             },
         )?;
-
-        Ok(())
+        Ok(self.get_swap_info_by_address(bitcoin_address)?.unwrap())
```

Comment:
Unwrap is bad, as it can panic on failure.

Reply:
True, made it in by mistake, we don't need it.